jacklemkus/jacklemkus/spiders/jacklemkus_spider.py

```diff
@@ -0,0 +1,111 @@
+from scrapy.spiders import CrawlSpider, Rule
+from scrapy.linkextractors import LinkExtractor
+
+
+class ItemSpider(CrawlSpider):
+    name = 'jacklemkus'
+    allowed_domains = ['jacklemkus.com']
+    start_urls = ['https://www.jacklemkus.com']
+
+    rules = [
+        Rule(LinkExtractor(restrict_xpaths='//*[@id="nav"]'), callback='parse'),
+        Rule(LinkExtractor(restrict_xpaths='//*[@id="products-grid"]'), callback='parse_product_page')
+    ]
+
+    def parse(self, response):
+        last_page = response.css("div.js-infinite-scroll-pager-data::attr(data-lastpage)").get()
+        if last_page:
+            for page_number in range(1, int(last_page)+1):
+                url = response.url + f"?p={page_number}"
+                yield response.follow(url, callback=self._parse)
+
+    def parse_product_page(self, response):
+        product_page_parser = ProductPageParser(response)
+        product_page_parser.fetch_product_info()
+        yield product_page_parser.product_info
+
+class ProductPageParser:
+    def __init__(self, response) -> None:
+        self.product_info = {}
+        self.response = response
```

Comment:
response is needed in multiple methods so i made it a data member.
product_info can be removed

Reply:
Still, we do not need to do that. since we don't have a deep level of coupling. we can pass the response to parse method and then to next method where needed.